File: office-space-frontend/src/main/web/office-space-web/src/app/services/cluster.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TableSearchRequest } from '../dtos/tableSearch';
import { StorageService } from './auth/storage.service';

@Injectable({
  providedIn: 'root'
})
export class ClusterService {

  readonly url = `${this.storage.get(StorageService.API)}cluster`;

  constructor(private http: HttpClient, private storage: StorageService) { }

  create(data): Observable<any> {
    return this.http.post(`${this.url}/create`, data);
  }

  getListView(data: TableSearchRequest): Observable<any> {
    return this.http.post(`${this.url}/list-view`, data);
  }

  update(data): Observable<any> {
    return this.http.patch(this.url, data);
  }
}

File: office-space-backend/src/main/java/com/ourfancyteamname/officespace/services/ClusterService.java
package com.ourfancyteamname.officespace.services;

import org.springframework.data.domain.Page;

import com.ourfancyteamname.officespace.db.entities.ProcessCluster;
import com.ourfancyteamname.officespace.dtos.ProcessGeneralDto;
import com.ourfancyteamname.officespace.dtos.TableSearchRequest;

public interface ClusterService {

  ProcessCluster create(ProcessGeneralDto processGeneralDto);

  Page<ProcessGeneralDto> getListView(TableSearchRequest tableSearchRequest);

  ProcessCluster update(ProcessGeneralDto processGeneralDto);
}
